### PR TITLE
improve methods concepts

### PR DESCRIPTION
**Pull request by DickBaker**
[improve methods concepts (NRT, analyzer tips etc)](https://github.com/dotnet/docs/commit/0ff7d9d5ae8ccb620d78d956a8acc742fec498dc) 

KISS+YAGNI: remove excess usings, private clutter
now C#11 and .NET 8 LTS GA'ed, can remove `<LangVersion>`

### CODE REVIEW

File: samples/snippets/csharp/concepts/methods/byref106.cs [hunk 1]
```diff
@@ -1,22 +1,20 @@
...
+        Console.WriteLine("Back in Main, value = {0}", value);
+    }
 
-   static void ModifyValue(ref int i)
-   {
-      i = 30;
-      Console.WriteLine("In ModifyValue, parameter value = {0}", i);
-      return;
-   }
+    static void ModifyValue(ref int i)
```
Comment (BillWagner) [inline samples/snippets/csharp/concepts/methods/byref106.cs:12]: add access modifier:

```suggestion
    private static void ModifyValue(ref int i)
```

**Review by BillWagner — APPROVED** [review]
Also a few small nits, and then this is ready.

Thanks again @DickBaker

File: samples/snippets/csharp/concepts/methods/async1.cs [hunk 2]
```diff
@@ -1,19 +1,12 @@
...
-    public static void Main()
-    {
-       DoSomethingAsync().Wait();
-    }
+    public static void Main() => DoSomethingAsync().Wait();
 
     private static async Task DoSomethingAsync()
     {
-        int result = await DelayAsync();
+        var result = await DelayAsync();
```
Comment (gfoidl) [inline samples/snippets/csharp/concepts/methods/async1.cs:9]: @BillWagner do we want to use `var` here or the concrete types?

E.g. in dotnet/runtime it must be the concrete type, except it's clearly visible what type it is.

Comment (BillWagner) [reply samples/snippets/csharp/concepts/methods/async1.cs:9]: Yes, it should be the concrete type. I missed that in my review.

File: samples/snippets/csharp/concepts/methods/async1.cs [hunk 3]
```diff
@@ -1,23 +1,16 @@
...
 {
     // This Click event is marked with the async modifier.
-    public static void Main()
-    {
-       DoSomethingAsync().Wait();
-    }
+    public static void Main() => DoSomethingAsync().Wait();
 
-    private static async Task DoSomethingAsync()
+    static async Task DoSomethingAsync()
```
[the remark this answers is not in the document]
Comment (DickBaker) [reply samples/snippets/csharp/concepts/methods/async1.cs]: @BillWagner sorry about those nits (incl Günther's catch)
although I dislike the boring private accessor keyword I accept your mantra, so won't PR those again

**Pull request merged**